agent/config.go

```diff
@@ -1167,6 +1174,10 @@ func DecodeConfig(r io.Reader) (*Config, error) {
 		fmt.Fprintln(os.Stderr, "==> DEPRECATION: atlas_endpoint is deprecated and "+
 			"is no longer used. Please remove it from your configuration.")
 	}
+	if len(result.DeprecatedHTTPAPIResponseHeaders) > 0 {
```

Comment:
When you move the mangling to `DecodeConfig` you can move the log statement as well. I'll take care of the other ones later.